opm/simulators/flow/FlowProblem.hpp

```diff
@@ -2212,361 +1608,16 @@ class FlowProblem : public GetPropType<TypeTag, Properties::BaseProblem>
-        const int restart_step = initconfig.getRestartStep();
-        {
-            simulator.setTime(schedule.seconds(restart_step));
-
-            simulator.startNextEpisode(simulator.startTime() + simulator.time(),
-                                       schedule.stepLength(restart_step));
-            simulator.setEpisodeIndex(restart_step);
-        }
-        eclWriter_->beginRestart();
-
-        Scalar dt = std::min(eclWriter_->restartTimeStepSize(), simulator.episodeLength());
-        simulator.setTimeStepSize(dt);
-
-        std::size_t numElems = this->model().numGridDof();
-        initialFluidStates_.resize(numElems);
-        if constexpr (enableSolvent) {
-            this->solventSaturation_.resize(numElems, 0.0);
-            this->solventRsw_.resize(numElems, 0.0);
-        }
-
-        if constexpr (enablePolymer)
-            this->polymer_.concentration.resize(numElems, 0.0);
-
-        if constexpr (enablePolymerMolarWeight) {
-            const std::string msg {"Support of the RESTART for polymer molecular weight "
-                                   "is not implemented yet. The polymer weight value will be "
-                                   "zero when RESTART begins"};
-            OpmLog::warning("NO_POLYMW_RESTART", msg);
-            this->polymer_.moleWeight.resize(numElems, 0.0);
-        }
-
-        if constexpr (enableMICP) {
-            this->micp_.resize(numElems);
-        }
-
-        // Initialize mixing controls before trying to set any lastRx valuesx
-        this->mixControls_.init(numElems, restart_step, eclState.runspec().tabdims().getNumPVTTables());
-
-        for (std::size_t elemIdx = 0; elemIdx < numElems; ++elemIdx) {
-            auto& elemFluidState = initialFluidStates_[elemIdx];
-            elemFluidState.setPvtRegionIndex(pvtRegionIndex(elemIdx));
-            eclWriter_->outputModule().initHysteresisParams(simulator, elemIdx);
-            eclWriter_->outputModule().assignToFluidState(elemFluidState, elemIdx);
-
-            // Note: Function processRestartSaturations_() mutates the
-            // 'ssol' argument--the value from the restart file--if solvent
-            // is enabled.  Then, store the updated solvent saturation into
-            // 'solventSaturation_'.  Otherwise, just pass a dummy value to
-            // the function and discard the unchanged result.  Do not index
-            // into 'solventSaturation_' unless solvent is enabled.
-            {
-                auto ssol = enableSolvent
-                    ? eclWriter_->outputModule().getSolventSaturation(elemIdx)
-                    : Scalar(0);
-
-                processRestartSaturations_(elemFluidState, ssol);
-
-                if constexpr (enableSolvent) {
-                    this->solventSaturation_[elemIdx] = ssol;
-                    this->solventRsw_[elemIdx] = eclWriter_->outputModule().getSolventRsw(elemIdx);
-                }
-            }
+    virtual void readEclRestartSolution_() = 0;
```

Comment:
this is only called in FlowProblemBlackoil so there is no need for this virtual, at least not as-is.